### PR TITLE
Course Filter Options, Various UX/UI improvements

### LINKED ISSUE #54: Content not filling when not enough data

### PR DESCRIPTION
**Pull request by aomi**
## Overview
Does a number of UI/UX improvements, code-refactoring etc.
Closes #54 

### Changes
- Card main and sub-text font size has been increased.
- Cards are not rounded and margin between cards have been decreased.
- Cards now have a green gradient on hover and a blue gradient on select.
- Course title/subject/code is displayed prior to description, sections for course loading.
- Top bar has a filter popdown menu.
- Filtering to only show courses in session is now an option.
- Top bar uses a breadcrumb to show which subject currently in. 
- CRN is now slightly smaller
- Section types besides labs, tutorials and lectures are handled now
- Rounded box shadow for section items
- Section type header is now smaller than the course subject/code

image

### CODE REVIEW

File: src/app/content/containers/Section.tsx [hunk 1]
```diff
@@ -46,44 +46,44 @@ export function SectionsContainer({ term, subject, code }: SectionsContainerProp
...
-        <>
-          <Divider />
-          <Heading size="2xl" color="black" my="2">
-            Tutorials
-          </Heading>
-          <Sections sections={tutorials} seats={seats} />
-        </>
-      )}
+      {categorizedSections.map((c) => {
+        if (c.sections && c.sections.length > 0) {
```
Comment (keithradford) [inline src/app/content/containers/Section.tsx:74]: cool change 😎

File: src/app/sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -5,7 +5,6 @@ import { SelectedCourse } from '../../App';
 import { Course, KualiSubject } from '../../fetchers';
```
Comment (keithradford) [inline src/app/sidebar/Sidebar.tsx:5]: the import for `useState` should probably be removed (couldn't add a comment there)

File: src/app/sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -19,6 +18,9 @@ export interface SidebarProps {
 
   selectedCourse?: SelectedCourse;
   setSelectedCourse: Dispatch<SetStateAction<SelectedCourse | undefined>>;
+
+  selectedSubject?: string;
+  setSelectedSubject: Dispatch<SetStateAction<string | undefined>>;
```
Comment (keithradford) [inline src/app/sidebar/Sidebar.tsx:23]: maybe we should use callbacks instead of `Dispatch` now?

**Review by keithradford — APPROVED** [review]
A few things but overall I think these are good changes

File: src/app/sidebar/Sidebar.tsx (hunk 3 above)
Comment (aomi) [reply src/app/sidebar/Sidebar.tsx]: @keithradford check out the changes :+1:

**Pull request merged**